### PR TITLE
(CODEMGMT-308) Add hash keys to symbols middleware

### PR DESCRIPTION
**Pull request by austb**
Before this commit, API responses were received with hash keys as Strings.
With this commit, all hash responses will have their keys converted to
symbols if possible.

### CODE REVIEW

File: lib/puppet_forge/v3/base.rb [hunk 1]
```diff
@@ -78,8 +78,8 @@ def where(params)
         end
 
         # Return a paginated collection of all modules
-        def all
-          where(nil)
+        def all(params = {})
+          where(params)
```
Comment (austb) [inline lib/puppet_forge/v3/base.rb]: On top of using .all to get the entire paginated collection, v1 of the puppet_forge gem allowed the use of .all just like where, this change makes that functionality still work in v2.

File: lib/puppet_forge/v3/base.rb [hunk 2]
```diff
@@ -89,11 +89,18 @@ def get_collection(uri_path)
           new_collection(resp)
         end
 
+        # Faraday's Util#escape method will replace a '+' with '%2B' to prevent it being
+        # interpreted as a space. For compatibility with the Forge API, we would like a '+'
+        # to be interpreted as a space so they are changes to spaces here.
```
Comment (andersonmills) [inline lib/puppet_forge/v3/base.rb]: typo of changes => changed

File: lib/puppet_forge/v3/base.rb [hunk 3]
```diff
@@ -89,11 +89,18 @@ def get_collection(uri_path)
...
+        def unescape_plus(str)
```
Comment (andersonmills) [inline lib/puppet_forge/v3/base.rb]: This seems like an inaccurate name here. Can we call it convert_plus_to_space, or something like that?

**Comment by andersonmills** [comment]
Okay, then.  :+1:  for you!

**Pull request merged**